### ISSUE TITLE
Rollback interfering with Return action

### ISSUE THREAD

**Issue by Gouvernathor** [renpy/renpy#2994]
Sometimes, in a specific game of mine, in a screen with several buttons with Return actions returning different values, when clicking a button, then rolling back, then clicking another button, the value of the first button is affected to the _return variable the second time, instead of the value of the button having been clicked.

It doesn't always happen when calling that screen, but saving before or during the screen call, when encountering the bug, conserves it in a peculiar way.
The value of the first button being clicked on (say "a") is not saved, but the bug will always trigger when calling the screen from that save, then clicking on a button, then rolling back, then clicking another button.

Working to reproduce it in a clean project. The screen is on layer "master", if it changes anything.

**Comment by Gouvernathor** [comment]
After adding a print statement inside the Return class, it appears the action is indeed called the second time, and returns the second value, but that value is somehow replaced by the first one before the call statement returns and renpy script continues to execute.

**Comment by Gouvernathor** [comment]
After examination, it appears that `renpy`(.export)`.in_fixed_rollback()` sometimes returns True even though `renpy.fix_rollback()` is never called in my game.

**Comment by renpytom** [comment]
Is this actually causing a problem? It's looking like an edge condition that can happen when Ren'Py runs out of rollback data.

**Comment by Gouvernathor** [comment]
Yep. I have an imagemap-like screen (actually a bunch of imagebuttons) and after a rollback the bug sometimes randomly happens, "fixing" the rollback in effect, with no apparent reason.

**Comment by Gouvernathor** [comment]
I'd be surprised if it's a out-of-data issue, because it's the `fixed_rollback_boundary` attribute of the `renpy.game.log` variable which finds itself with a value instead of None.
I'm still trying to isolate the exact moment where the actual value is affected, as the bug is very random and hard to reproduce.

I have no idea how that can have happened without a `renpy.fix_rollback()` call, so I guess your explaination is as valid as another. This is all very weird.
